apps/contracts/contracts/ProjectFactory.sol

```diff
@@ -1,46 +1,180 @@
 //SPDX-License-Identifier: UNLICENSED
-pragma solidity 0.8.9;
+pragma solidity 0.8.4;
 
-contract ProjectFactory {
+import {DataTypes} from "./libraries/DataTypes.sol";
+import "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+import "@semaphore-protocol/contracts/interfaces/ISemaphore.sol";
+import "@semaphore-protocol/contracts/interfaces/IVerifier.sol";
+import "@semaphore-protocol/contracts/base/SemaphoreCore.sol";
+import "@semaphore-protocol/contracts/base/SemaphoreGroups.sol";
+
+error InvalidTreeDepth();
+error InvalidTime();
+
+contract ProjectFactory is SemaphoreCore, SemaphoreGroups {
     
+    using EnumerableSet for EnumerableSet.UintSet;
+
+    address internal constant verifier = address(0);
+    uint8 internal constant MAX_DEPTH = 32;
+
+    uint256 internal _projectCount;
+    mapping(uint256 => DataTypes.Profile) internal profileInfo;
+    mapping(uint256 => DataTypes.Project) internal projectInfo;
```

Comment:
Having a question here, why would we need to save the id comment into a members array?